Cubical/ZCohomology/Groups/Wedge.agda

```diff
@@ -45,15 +54,66 @@ module _ {ℓ ℓ'} (A : Pointed ℓ) (B : Pointed ℓ') where
             λ f inker → helper ∣ f ∣₂ (I.Ker-i⊂Im-d 0 ∣ f ∣₂ inker)
     BijectionIso.surj bijIso p = I.Ker-Δ⊂Im-i 1 p (isContr→isProp (isContrHⁿ-Unit 0) _ _)
 
-  Hⁿ-⋁ (suc n) =
-    vSES→GroupIso _ _
-      (ses (isOfHLevelSuc 0 (isContrHⁿ-Unit n))
-           (isOfHLevelSuc 0 (isContrHⁿ-Unit (suc n)))
-           (I.d (suc n))
-           (I.Δ (suc (suc n)))
-           (I.i (suc (suc n)))
-           (I.Ker-i⊂Im-d (suc n))
-           (I.Ker-Δ⊂Im-i (suc (suc n))))
+  Hⁿ-⋁ (suc n) = Iso+Hom→GrIso mainIso
+                                (sElim2 (λ _ _ → isOfHLevelPath 2 (isOfHLevel× 2 setTruncIsSet setTruncIsSet) _ _)
+                                         λ _ _ → refl)
+    where
+    helpIso : ∀ {ℓ'''} {C : Type ℓ'''} → Iso (A ⋁ B → C) (Σ[ f ∈ (typ A → C) × (typ B → C) ] (fst f) (pt A) ≡ (snd f) (pt B))
+    Iso.fun helpIso f = ((λ x → f (inl x)) , λ x → f (inr x)) , cong f (push tt)
+    Iso.inv helpIso ((f , g) , p) (inl x) = f x
+    Iso.inv helpIso ((f , g) , p) (inr x) = g x
+    Iso.inv helpIso ((f , g) , p) (push a i) = p i
+    Iso.rightInv helpIso ((f , g) , p) = ΣPathP (ΣPathP (refl , refl) , refl)
+    Iso.leftInv helpIso f = funExt λ {(inl a) → refl ; (inr a) → refl ; (push a i) → refl}
```

Comment:
A lot of redundant parentheses here

Reply:
I actually seem to need all of them since (_,_) is right-associative. I could change the alter the statement slightly to avoid this, but then I would either have to juggle with truncations more than I do now or make the lemma about truncating sigma types (the one I called moveSetTruncΣ) much less general.

Reply:
I mean the ones in the pattern-matching:
```
    Iso.leftInv helpIso f = funExt λ {(inl a) → refl ; (inr a) → refl ; (push a i) → refl}
```

It should be possible to write:

```
    Iso.leftInv helpIso f = funExt λ { inl a → refl ; inr a → refl ; push a i → refl}
```

Alternatively
```
    Iso.leftInv helpIso f = funExt λ { _ → refl }
```
might also work?

Reply:
Neither of those things work:-( Do these things work for you? Maybe I don't have the latest version of Agda.

Reply:
Ok, nevermind this then. I rarely use pattern-matching lambdas except for hcomp so I guess I just don't know what is syntactically allowed